Problem with exitcodes in launch.m

I am getting,  in `ustar_cp_refactor_main`:
```
E           matlab.engine.MatlabExecutionError: 
E             File /Users/dorchard/Documents/iccs/ONEFlux/oneflux_steps/ustar_cp_refactor_wip/launch.m, line 166, in launch
E           Output argument "exitcode" (and possibly others) not assigned a value in the execution with "saveResult" function.

/opt/homebrew/lib/python3.11/site-packages/matlab/engine/fevalfuture.py:76: MatlabExecutionError
```

--- a/oneflux_steps/ustar_cp_refactor_wip/launch.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch.m
@@ -42,10 +42,11 @@ function exitcode = launch(input_folder, output_folder)
         
         dataset = textscan(fid,'%[^\n]');dataset = dataset{1};
         
-        [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, notes] ...
+        [errorCode, site, year, lat, lon, timezone, htower, timeres, sc_negl, notes] ...
             = notValidHeader(dataset);
 
-        if exitcode == 1
+        if errorCode == 1
+            exitcode = 1;
             continue;
         end
         
@@ -70,15 +71,17 @@ function exitcode = launch(input_folder, output_folder)
         [header, data, columns_index] = loadData(input_folder, filename, notes, input_columns_names);
         
         % parse header, by alessio
-        [exitcode, columns_index] = mapColumnNamesToIndices(header, input_columns_names, notes, columns_index);
+        [errorCode, columns_index] = mapColumnNamesToIndices(header, input_columns_names, notes, columns_index);
         
-        if exitcode == 1
+        if errorCode == 1
+            exitcode = 1;
             continue;
         end
         
-        [ppfd_from_rg, exitcode] = ppfdColExists(PPFD_INDEX, columns_index, input_columns_names);
+        [ppfd_from_rg, errorCode] = ppfdColExists(PPFD_INDEX, columns_index, input_columns_names);
         
-        if exitcode == 1
+        if errorCode == 1
+            exitcode = 1;
             continue;
         end
         
@@ -87,7 +90,9 @@ function exitcode = launch(input_folder, output_folder)
         Ta = data(:, columns_index(TA_INDEX));
         Rg = data(:, columns_index(RG_INDEX));
 
-        [PPFD, ppfd_from_rg] = areAllPpfdValuesInvalid(ppfd_from_rg, columns_index, PPFD_INDEX, data);
+        %ppfd_from_rg == 0 can be evaluated outside the function below
+        [PPFD, ppfd_from_rg] = areAllPpfdValuesInvalid(ppfd_from_rg, columns_index, PPFD_INDEX, data); 
+
         
         if 1 == ppfd_from_rg
 
@@ -99,8 +104,9 @@ function exitcode = launch(input_folder, output_folder)
 
         [uStar, NEE, Ta, PPFD, Rg] = setMissingDataNan(uStar, NEE, Ta, PPFD, Rg);
 
-        [exitcode] = anyColumnsEmpty(uStar, NEE, Ta, Rg);
-        if exitcode == 1
+        [errorCode] = anyColumnsEmpty(uStar, NEE, Ta, Rg);
+        if errorCode == 1
+            exitcode = 1;
             continue;
         end
 
@@ -163,14 +169,17 @@ function exitcode = launch(input_folder, output_folder)
             = cpdAssignUStarTh20100901(Stats2,fPlot,cSiteYr); 
         
 
-        [error_str, cSiteYr, exitcode] = saveResult(cFailure, cSiteYr, output_folder, site, year, Cp, clock, notes);
-
+        [error_str, cSiteYr, errorCode] = saveResult(cFailure, cSiteYr, output_folder, site, year, Cp, clock, notes);
+        if errorCode == 1
+            exitcode = 1;
+            continue;
+        end
         % by alessio
         %end    
     %     print -djpeg100 Plot2_4Season_CACa1-2001; 
 
         
-
+        clear global globalFrequencyMetadata
         clear uStar cFailure cMode cSiteYr fNight fPlot fSelect n nBoot sSine t 
         clear tW Cp3 CpW FracModeD FracSelect FracSig NEE PPFD Rg Stats2 Stats3 T Ta Cp Cp2
     end

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/anyColumnsEmpty.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/anyColumnsEmpty.m
@@ -1,28 +1,28 @@
-function [exitcode] = anyColumnsEmpty(uStar, NEE, Ta, Rg)
+function [errorCode] = anyColumnsEmpty(uStar, NEE, Ta, Rg)
 
-    exitcode = 0;
+    errorCode = 0;
 
     %         data_L3(data_L3==-9999)=NaN; data_L3(data_L3==-6999)=NaN;
 
     % by carlo, added by alessio on February 21, 2014
     if sum(isnan(NEE)) == numel(NEE); 
         fprintf('NEE is empty!\n');
-        exitcode = 1
+        errorCode = 1;
         return;
     end
     if sum(isnan(uStar)) == numel(uStar); 
         fprintf('uStar is empty!\n');
-        exitcode = 1
+        errorCode = 1;
         return;
     end
     if isempty(Ta); 
         fprintf('Ta is empty!\n');
-        exitcode = 1
+        errorCode = 1;
         return;
     end
     if isempty(Rg); 
         fprintf('Rg is empty!\n');
-        exitcode = 1
+        errorCode = 1;
         return;
     end
     

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/areAllPpfdValuesInvalid.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/areAllPpfdValuesInvalid.m
@@ -1,5 +1,5 @@
 function [PPFD, ppfd_from_rg] = areAllPpfdValuesInvalid(ppfd_from_rg, columns_index, PPFD_INDEX, data, varargin)
-    
+    PPFD = '';
     if length(varargin) > 0
         data = importdata(varargin{1}, ',');
     end

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/mapColumnNamesToIndices.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/mapColumnNamesToIndices.m
@@ -1,4 +1,4 @@
-function [exitcode, columns_index] = ...
+function [errorCode, columns_index] = ...
     mapColumnNamesToIndices(header, input_columns_names, notes, columns_index, varargin)
 
     if length(varargin) > 0
@@ -10,7 +10,7 @@ function [exitcode, columns_index] = ...
 
     end
     on_error = 0;
-    exitcode = 0;
+    errorCode = 0;
 
     for y = 1:length(header(9+length(notes),:))
         for i = 1:numel(input_columns_names)
@@ -30,7 +30,7 @@ function [exitcode, columns_index] = ...
     end
     
     if 1 == on_error 
-        exitcode = 1;
+        errorCode = 1;
         return;
     end
 end

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/notValidHeader.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/notValidHeader.m
@@ -1,12 +1,12 @@
-function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, notes] ... 
+function [errorCode, site, year, lat, lon, timezone, htower, timeres, sc_negl, notes] ... 
     = notValidHeader(dataset)
-    exitcode = 0; 
+    errorCode = 0; 
     [site, year, lat, lon, timezone, htower, timeres, sc_negl, notes] = deal(NaN);
     % get site
     r = strncmpi(dataset(1), 'site', 4);
     if 0 == r
         fprintf('site keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     site = dataset(1);
@@ -16,7 +16,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(2), 'year', 4);
     if 0 == r
         fprintf('year keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     year = dataset(2);
@@ -26,7 +26,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(3), 'lat', 3);
     if 0 == r
         fprintf('lat keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     lat = dataset(3);
@@ -36,7 +36,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(4), 'lon', 3);
     if 0 == r
         fprintf('lon keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     lon = dataset(4);
@@ -46,7 +46,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(5), 'timezone', 8);
     if 0 == r
         fprintf('timezone keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     timezone = dataset(5);
@@ -56,7 +56,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(6), 'htower', 6);
     if 0 == r
         fprintf('htower keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     htower = dataset(6);
@@ -66,7 +66,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(7), 'timeres', 7);
     if 0 == r
         fprintf('timeres keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     timeres = dataset(7);
@@ -76,7 +76,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(8), 'Sc_negl', 7);
     if 0 == r
         fprintf('sc_negl keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     sc_negl = dataset(8);
@@ -86,7 +86,7 @@ function [exitcode, site, year, lat, lon, timezone, htower, timeres, sc_negl, no
     r = strncmpi(dataset(9), 'notes', 5);
     if 0 == r
         fprintf('notes keyword not found.\n');
-        exitcode = 1;
+        errorCode = 1;
         return;
     end       
     notes = dataset(9);

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/ppfdColExists.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/ppfdColExists.m
@@ -1,6 +1,6 @@
-function [ppfd_from_rg, exitcode] = ppfdColExists(PPFD_INDEX, columns_index, input_columns_names)
+function [ppfd_from_rg, errorCode] = ppfdColExists(PPFD_INDEX, columns_index, input_columns_names)
 
-    exitcode = 0;
+    errorCode = 0;
     on_error = 0;
     ppfd_from_rg = 0;
     for i = 1:numel(columns_index)
@@ -15,7 +15,7 @@ function [ppfd_from_rg, exitcode] = ppfdColExists(PPFD_INDEX, columns_index, inp
     end
     
     if 1 == on_error
-        exitcode = 1;
+        errorCode = 1;
         return;
     end
 

--- a/oneflux_steps/ustar_cp_refactor_wip/launch_functions/saveResult.m
+++ b/oneflux_steps/ustar_cp_refactor_wip/launch_functions/saveResult.m
@@ -1,5 +1,6 @@
-function [error_str, cSiteYr, exitcode] = saveResult(cFailure, cSiteYr, output_folder, site, year, Cp, clock, notes)
+function [error_str, cSiteYr, errorCode] = saveResult(cFailure, cSiteYr, output_folder, site, year, Cp, clock, notes)
     error_str = "";
+    errorCode = 0;
     if isempty(cFailure)
         %save([cOut,'4-season_analysis_',strrep(cSiteYr,'.txt',''),'.mat'],'Cp2','Stats2','Cp3','Stats3',...
         %    'Cp','n','tW','CpW','cMode','cFailure','fSelect','sSine','FracSig','FracModeD','FracSelect');
@@ -18,6 +19,6 @@ function [error_str, cSiteYr, exitcode] = saveResult(cFailure, cSiteYr, output_f
     else
         error_str = [error_str; char(site),'_uscp_',char(year),' ',cFailure];
         fprintf('%s\n', cFailure);
-        exitcode = 1;
+        errorCode = 1;
     end
 end

